crates/oxc_linter/src/rules/react/require_render_return.rs

```diff
@@ -52,69 +49,116 @@ declare_oxc_lint!(
 );
 
 impl Rule for RequireRenderReturn {
-    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
-        if !is_es5_component(node) && !is_es6_component(node) {
-            return;
-        }
+    fn run_once(&self, ctx: &LintContext) {
+        let mut render_fn_set: FxHashSet<AstNodeId> = FxHashSet::default();
 
-        match node.kind() {
-            AstKind::Class(cl) => {
-                if let Some((fn_body, is_arrow_function_expression)) =
-                    cl.body.body.iter().find_map(|ce| match ce {
-                        ClassElement::MethodDefinition(md)
-                            if md.key.is_specific_static_name("render") =>
-                        {
-                            md.value.body.as_ref().map(|v| (v, false))
-                        }
-                        ClassElement::PropertyDefinition(pd)
-                            if pd.key.is_specific_static_name("render") =>
-                        {
-                            if let Some(Expression::ArrowFunctionExpression(ref ae)) = pd.value {
-                                Some((&ae.body, ae.expression))
-                            } else {
-                                None
-                            }
-                        }
-                        _ => None,
-                    })
-                {
-                    if is_arrow_function_expression || has_return_in_fn_body(fn_body) {
-                        return;
-                    }
-
-                    ctx.diagnostic(require_render_return_diagnostic(fn_body.span));
-                }
+        for node in ctx.nodes().iter() {
```

Comment:
Is it possible to not iterate through nodes in in `run_once` for correctness rules?